src/convertToObject.js

```diff
@@ -6,7 +6,22 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylePrev = sourceString.trim().split(';');
+
+  const style = stylePrev
+    .filter((elm) => {
+      if (elm.indexOf(':') !== -1) {
+        return elm;
+      }
+    })
+    .map((elm) => {
+      return elm.split(':');
+    })
+    .reduce((object, elm) => {
+      return { ...object, [elm[0].trim()]: elm[1].trim() };
```

Comment:
Using the spread operator inside the reduce callback is not efficient as it creates a new object on each iteration. Consider using the accumulator to directly assign the new properties.